src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_yx_fsv16_1x1.cpp

```diff
@@ -215,47 +229,222 @@ JitConstants ConvolutionKernel_b_fs_yx_fsv16_1x1::GetJitConstants(const convolut
+                        finalKernelName,
+                        jit,
+                        entryPoint,
+                        EXE_MODE_DEFAULT,
+                        true,
+                        !newParams.bias.empty(),
+                        1, 0, 1,
+                        newParams.is_shape_agnostic);
+
+        if (newParams.deformable_mode) {
+            kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 1});
+            if (newParams.deformable_mask_enabled)
+                kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 2});
+        }
+
+        if (!newParams.weights_zero_points.empty())
+            kernel.params.arguments.push_back({ArgumentDescriptor::Types::WEIGHTS_ZERO_POINTS, 1});
+        if (!newParams.activations_zero_points.empty())
+            kernel.params.arguments.push_back({ArgumentDescriptor::Types::ACTIVATIONS_ZERO_POINTS, 1});
+        if (!newParams.compensation.empty())
+            kernel.params.arguments.push_back({ArgumentDescriptor::Types::COMPENSATION, 1});
+
+        uint32_t fused_deps_total = 0;
+        for (auto& fused_dep : newParams.fused_ops) {
+            for (int i = 0; i < static_cast<int>(fused_dep.dep_size); i++) {
+                kernel.params.arguments.push_back({ ArgumentDescriptor::Types::INPUT_OF_FUSED_PRIMITIVE, fused_deps_total });
+                fused_deps_total++;
+            }
+        }
+    }
+    kd.autoTuneIndex = -1;
+
+    return {kd};
+ }
+
+void ConvolutionKernel_b_fs_yx_fsv16_1x1::GetUpdateDispatchDataFunc(KernelData& kd) const {
+    if (kd.kernels.size() == 1) {
+        Parent::GetUpdateDispatchDataFunc(kd);
+    } else {
+        kd.update_dispatch_data_func = [this](const Params& params, KernelData& kd) {
+            const auto& prim_params = static_cast<const convolution_params&>(params);
+            auto dispatchData = SetDefault(prim_params);
+            size_t execute_kernel_idx = 3;
+            if (dispatchData.cldnnStyle.blockWidth == 1) {
+                execute_kernel_idx = 0;
+            } else if (dispatchData.cldnnStyle.blockWidth == 2) {
+                execute_kernel_idx = 1;
+            } else if (dispatchData.cldnnStyle.blockWidth == 4) {
+                execute_kernel_idx = 2;
+            }
+            for (size_t i = 0; i < kd.kernels.size(); i++) {
+                kd.kernels[i].params.workGroups.global = dispatchData.gws;
+                kd.kernels[i].params.workGroups.local = dispatchData.lws;
+                if (execute_kernel_idx == i) {
+                    kd.kernels[i].skip_execution = KernelData::SkipKernelExecution(prim_params);
+                } else {
+                    kd.kernels[i].skip_execution = true;
+                }
+            }
+            kd.internalBufferSizes.clear();
+            kd.internalBufferSizes.push_back(prim_params.inputs[0].PhysicalSizeInBytes());
+            kd.internalBufferDataType = prim_params.inputs[0].GetDType();
```

Comment:
Why do we need this internal buffer?

Reply:
Applied, frankly no need, it came from convolution_kernel_base.